### PR TITLE
Update set_block functionality and remove eigen

### PR DESCRIPTION
**Pull request by jorgensd**
Fixes changes introduced in https://github.com/FEniCS/dolfinx/pull/1417 and https://github.com/FEniCS/dolfinx/pull/1493

Removes eigen in favor of xtensor

### CODE REVIEW

File: src/mesh.cpp [hunk 1]
```diff
@@ -141,8 +141,8 @@ create_spoke_mesh(MPI_Comm comm, std::size_t target_dofs,
                     {0, 2, 3, 4}, {6, 7, 4, 2}, {2, 3, 4, 7}};
 
   // Calculate number of points and cells (only on process 0)
-  int npoints = 0;
-  int ncells = 0;
+  std::uint32_t npoints = 0;
```
Comment (chrisrichardson) [inline src/mesh.cpp:144]: I know this is unsigned, but we've generally been using signed ints for consistency everywhere.

**Review by IgorBaratta — APPROVED** [review]

**Comment by jorgensd** [comment]
The unstructured mesh doesn't seem right. So we wait with merging until that has been fixed

**Comment by jorgensd** [comment]
Mesh issue fixed. However, it seems like the unstructured test is missing some boundary condition, as the solution is of magnitude 5e12
